semgrep-core/src/analyzing/AST_to_IL.ml

```diff
@@ -928,6 +973,69 @@ let rec stmt_aux env st =
   | G.OtherStmtWithStmt _ ->
       todo (G.S st)
 
+(* TODO: Maybe this and the following function could be merged *)
+and scrut_and_cases_to_exp env tok scrut_orig scrut cases =
+  (* If there is a scrutinee, the cases are expressions we need to check for equality with the scrutinee  *)
+  let ss, es =
+    List.fold_right
+      (fun case (ss, es) ->
+        match case with
+        | G.Case (tok, G.PatLiteral l) ->
+            ( ss,
+              {
+                e =
+                  Operator
+                    ( (G.Eq, tok),
+                      [ { e = Literal l; eorig = scrut_orig }; scrut ] );
+                eorig = scrut_orig;
+              }
+              :: es )
+        | G.Case (tok, G.OtherPat (_, [ E c ]))
+        | G.CaseEqualExpr (tok, c) ->
+            let c_ss, c' = expr_with_pre_stmts env c in
+            ( ss @ c_ss,
+              { e = Operator ((G.Eq, tok), [ c'; scrut ]); eorig = c } :: es )
+        | G.Default tok ->
+            (* Default should only ever be the final case, and cannot be part of a list of
+               `Or`ed together cases. It's handled specially in cases_and_bodies_to_stmts
+            *)
+            impossible (G.Tk tok)
+        | G.Case (tok, _) -> (ss, fixme_exp ToDo (G.Tk tok) scrut_orig :: es))
+      cases ([], [])
+  in
+  (ss, { e = Operator ((Or, tok), es); eorig = scrut_orig })
+
+and cases_to_exp env tok cases =
+  (* If we have no scrutinee, the cases are boolean expressions, so we Or them together *)
+  let ss, es =
+    List.fold_right
+      (fun case (ss, es) ->
+        match case with
+        | G.Case (_, G.PatLiteral l) ->
+            ( ss,
+              (* TODO: seems bad to make an artificial eorig, but seems to be nothing to use  *)
```

Comment:
In similar cases we pass the `eorig` as an argument to the function.

Reply:
Hmmm, but yeah, we have no scrutiny so... I also don't know what else to do. 😄 Maybe it should be an optional.